nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -28,11 +28,23 @@ export async function storeCommitment(commitment, nsk) {
     isOnChain: Number(commitment.isOnChain) || -1,
     isPendingNullification: false, // will not be pending when stored
     isNullified: commitment.isNullified,
-    isNullifiedOnChain: Number(commitment.isNullifiedOnChain),
+    isNullifiedOnChain: Number(commitment.isNullifiedOnChain) || -1,
     nullifier: nullifierHash,
     blockNumber: -1,
   };
-  return db.collection(COMMITMENTS_COLLECTION).insertOne(data);
+  // a chain reorg may cause an attempted overwrite. We should allow this, hence
+  // the use of replaceOne.
+  return db
```

Comment:
The `storeCommitment` call that we are trying to guard for here is the one in `blockProposed` handler. However, I believe it will only get called in the `else` condition of `countCommitments(transaction.commitments)) > 0`. That is, we only call `storeCommitment` if we do not have the commitment already in our local DB.

Is there a situation i've missed where we actually replace an existing document with this call?

Reply:
Yes, you're quite right.  This was an unnecessary protection.  I've removed it